optimism/src/keccak/trace.rs

```diff
@@ -56,7 +56,26 @@ impl<F: Field> CircuitTrait<ZKVM_KECCAK_COLS, Steps, F, KeccakEnv<F>> for Keccak
         });
     }
 
-    fn pad(&mut self, step: Steps) -> bool {
+    fn pad_with_row(&mut self, step: Steps, row: &[F; ZKVM_KECCAK_COLS]) -> bool {
```

Comment:
For push_row, line 52:
I would say what happens when the index is bigger. If it has to fail with an exception, it must be explicit.

Reply:
Could you also comments what these functions are supposed to do, please? The name, IMO, is not explicit enough. In particular, the condition `if let Round(_) = step` is not obvious.

Reply:
Never mind, I will add more comments and improve the readability of it.